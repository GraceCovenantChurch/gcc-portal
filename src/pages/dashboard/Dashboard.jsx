import React, { Component } from "react";
import clsx from "clsx";
import PropTypes from "prop-types";

import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Paper from "@material-ui/core/Paper";

import Chart from "./Chart";

const styles = (theme) => ({
  paper: {
    padding: theme.spacing(2),
    display: "flex",
    overflow: "auto",
    flexDirection: "column",
  },
  fixedHeight: {
    height: 240,
  },
});

const data = [
  createData("00:00", 0),
  createData("03:00", 300),
  createData("06:00", 600),
  createData("09:00", 800),
  createData("12:00", 1500),
  createData("15:00", 2000),
  createData("18:00", 2400),
  createData("21:00", 2400),
  createData("24:00", undefined),
];

// Generate Sales Data
function createData(time, amount) {
  return { time, amount };
}

class Dashboard extends Component {
  render() {
    let { classes } = this.props;
    const fixedHeightPaper = clsx(classes.paper, classes.fixedHeight);

    return (
      <Grid container spacing={3}>
        <Grid item xs={12} md={12} lg={12}>
          <Paper className={fixedHeightPaper}>
            <Chart data={data} />
          </Paper>
        </Grid>
      </Grid>
    );
  }
}

Dashboard.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Dashboard);
