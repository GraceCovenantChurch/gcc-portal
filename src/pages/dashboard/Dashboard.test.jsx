import React from "react";
import { shallow } from "enzyme";
import Dashboard from "./Dashboard";
import renderer from "react-test-renderer";

describe("Dashboard component", () => {
  let component, tree, wrapper;

  beforeEach(() => {
    component = renderer.create(<Dashboard />, { createNodeMock });
    tree = component.toJSON();

    wrapper = shallow(<Dashboard />);
  });

  it("UNIT: should be defined", () => {
    expect(Dashboard).toBeDefined();
  });

  it("SNAPSHOT: should render correctly", () => {
    expect(tree).toMatchSnapshot();
  });
});
