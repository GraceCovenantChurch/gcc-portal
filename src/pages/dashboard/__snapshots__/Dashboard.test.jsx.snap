// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component SNAPSHOT: should render correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12"
  >
    <div
      className="MuiPaper-root Dashboard-paper-1 Dashboard-fixedHeight-2 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <h1
        className="MuiTypography-root MuiTypography-h2 MuiTypography-colorPrimary MuiTypography-gutterBottom MuiTypography-alignCenter"
      >
        Today
      </h1>
      <div
        className="recharts-responsive-container"
        id="undefined"
        style={
          Object {
            "height": "100%",
            "maxHeight": undefined,
            "minHeight": undefined,
            "minWidth": undefined,
            "width": "100%",
          }
        }
      >
        <div />
      </div>
    </div>
  </div>
</div>
`;
